src/browser/modules/D3Visualization/lib/visualization/utils/pairwiseArcsRelationshipRouting.ts

```diff
@@ -21,20 +21,30 @@ import measureText from './textMeasurement'
 import LoopArrow from './loopArrow'
 import StraightArrow from './straightArrow'
 import ArcArrow from './arcArrow'
+import GraphStyle from 'browser/modules/D3Visualization/graphStyle'
 
 export default class PairwiseArcsRelationshipRouting {
-  style: any
-  constructor(style: any) {
+  style: GraphStyle
+  canvas: HTMLCanvasElement
+  constructor(style: GraphStyle) {
     this.style = style
+    this.canvas = document.createElement('canvas')
```

Comment:
Just a note: All the text measurement before seemed to reuse the same canvas, rather than create their own: `d3.select('canvas#textMeasurementCanvas)`.  I'm assuming this is a convenience tradeoff you've made that just uses a bit more memory but isn't a problem?

Reply:
@OskarDamkjaer - that's correct. It's not the bottleneck. Even I don't do the refactor, it should be fine.

Reply:
@OskarDamkjaer - `getContext` needs some time:
<img width="583" alt="Screenshot 2021-12-21 at 22 52 53" src="https://user-images.githubusercontent.com/26452483/147008042-2c7af0c8-7894-4786-97f3-875e2b7d371a.png">
So, it's slightly more efficient to pass the same `context` to the method to save time.

But, again, this part is not the bottleneck. Even we don't optimise it, it won't affect the performance too much.